flytekit/common/tasks/notebook.py

```diff
@@ -0,0 +1,542 @@
+                cpu_request=cpu_request,
+                gpu_request=gpu_request,
+                memory_request=memory_request,
+                storage_limit=storage_limit,
+                cpu_limit=cpu_limit,
+                gpu_limit=gpu_limit,
+                memory_limit=memory_limit,
+                environment=environment
+            )
+        )
+
+    @_exception_scopes.system_entry_point
+    def unit_test(self, **input_map):
+        """
+        :param dict[Text, T] input_map: Python Std input from users.  We will cast these to the appropriate Flyte
+            literals.
+        :returns: Depends on the behavior of the specific task in the unit engine.
+        """
+        return _engine_loader.get_engine('unit').get_task(self).execute(
+            _type_helpers.pack_python_std_map_to_literal_map(input_map, {
+                k: _type_helpers.get_sdk_type_from_literal_type(v.type)
+                for k, v in _six.iteritems(self.interface.inputs)
+            })
+        )
+
+    @_exception_scopes.system_entry_point
+    def local_execute(self, **input_map):
+        """
+        :param dict[Text, T] input_map: Python Std input from users.  We will cast these to the appropriate Flyte
+            literals.
+        :rtype: dict[Text, T]
+        :returns: The output produced by this task in Python standard format.
+        """
+        return _engine_loader.get_engine('local').get_task(self).execute(
+            _type_helpers.pack_python_std_map_to_literal_map(input_map, {
+                k: _type_helpers.get_sdk_type_from_literal_type(v.type)
+                for k, v in _six.iteritems(self.interface.inputs)
+            })
+        )
+
+    @_exception_scopes.system_entry_point
+    def execute(self, context, inputs):
+        """
+        :param flytekit.engines.common.EngineContext context:
+        :param flytekit.models.literals.LiteralMap inputs:
+        :rtype: dict[Text, flytekit.models.common.FlyteIdlEntity]
+        :returns: This function must return a dictionary mapping 'filenames' to Flyte Interface Entities.  These
+            entities will be used by the engine to pass data from node to node, populate metadata, etc. etc..  Each
+            engine will have different behavior.  For instance, the Flyte engine will upload the entities to a remote
+            working directory (with the names provided), which will in turn allow Flyte Propeller to push along the
+            workflow.  Where as local engine will merely feed the outputs directly into the next node.
+        """
+        inputs_dict = _type_helpers.unpack_literal_map_to_sdk_python_std(inputs, {
+            k: _type_helpers.get_sdk_type_from_literal_type(v.type) for k, v in _six.iteritems(self.interface.inputs)
+        })
+
+        # Execute Notebook via Papermill.
+        input_notebook_path = self._notebook_path
+        output_notebook_path = input_notebook_path[:len(input_notebook_path) - 6] + '-out' + input_notebook_path[len(
```

Comment:
can we guarantee this `- 6`? what does it represent?

Reply:
i presume `.ipynb`? will ipython run a notebook i give named `mynotbook.notebook`? if yes, we should probably not make this assumption about path and instead just append `.flyteout`.  alternatively, we could explicitly enforce that it must end in `.ipynb`--in which . case `.ipynb` should be a constant.

Reply:
replaced with just appending an `.out` . works fine